packages/@lwc/engine-core/src/framework/modules/static-parts.ts

```diff
@@ -105,3 +104,39 @@ export function applyStaticParts(
         applyRefs(part, owner);
     }
 }
+
+/**
+ * Reapplies static parts mounting to newly generated vnodes.
+ *
+ * @param n1 - the previous VStatic vnode
+ * @param n2 - the current VStatic vnode
+ * @param root - the root element
+ * @param renderer - the renderer to use
+ */
+export function patchStaticParts(n1: VStatic, n2: VStatic, root: Element, renderer: RendererAPI) {
+    if (!process.env.IS_BROWSER) {
+        return;
+    }
+
+    const { parts: currParts, owner: currPartsOwner } = n2;
+    if (isUndefined(currParts)) {
+        return;
+    }
+
+    const { parts: prevParts } = n1;
+    if (currParts.length !== prevParts?.length) {
+        // It should not be possible to reach this point as patching only occurs when the vnode is the same.
+        if (process.env.NODE_ENV !== 'production') {
+            throw new Error('Expected static parts to be the same for the same element');
+        }
+        // Recover from patching issue
+        return applyStaticParts(root, n2, renderer, false);
+    }
+
+    for (const [i, part] of currParts.entries()) {
```

Comment:
```suggestion
    for (for let i = 0; i < currParts.length; i++) {
      const part = currParts[i];
```

I only suggest micro-optimizations in cases where it might matter (like vdom code), which is why I suggest it here. Avoiding the extra array allocations should theoretically be a slight perf boost. (Perhaps even measurable by the Tachometer tests.)

Reply:
I was wondering about this and wasn't sure if we would want to use a for loop instead. 

I ran my benchmark with both versions and didn't see a different but it's a rudimentary test. 

I'll go with the for loop though as it'll be a perf boost!